src/main/java/se/sundsvall/oepintegrator/integration/opene/OpeneClientFactory.java

```diff
@@ -0,0 +1,171 @@
+package se.sundsvall.oepintegrator.integration.opene;
+
+import static java.util.Optional.ofNullable;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.zalando.problem.Status.INTERNAL_SERVER_ERROR;
+import static se.sundsvall.oepintegrator.integration.db.model.enums.IntegrationType.SOAP;
+
+import feign.Request;
+import feign.auth.BasicAuthRequestInterceptor;
+import feign.jaxb.JAXBContextFactory;
+import feign.soap.SOAPEncoder;
+import feign.soap.SOAPErrorDecoder;
+import jakarta.xml.soap.SOAPConstants;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.openfeign.FeignClientBuilder;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+import org.zalando.problem.Problem;
+import se.sundsvall.oepintegrator.integration.db.InstanceRepository;
+import se.sundsvall.oepintegrator.integration.db.model.InstanceEntity;
+import se.sundsvall.oepintegrator.integration.opene.rest.OpeneRestClient;
+import se.sundsvall.oepintegrator.integration.opene.soap.OpeneSoapClient;
+import se.sundsvall.oepintegrator.utility.EncryptionUtility;
+
+/**
+ * Factory class for creating OpenE clients. This class is responsible for creating and managing OpenE clients.
+ *
+ * <p>
+ * It creates clients for both REST and SOAP integrations and stores them in a map for easy access.
+ * </p>
+ * <p>
+ * It also provides methods for getting clients based on municipalityId and instanceId.
+ * </p>
+ */
+@Component
+public class OpeneClientFactory {
+
+	private static final Logger LOG = LoggerFactory.getLogger(OpeneClientFactory.class);
+
+	private static final JAXBContextFactory JAXB_FACTORY = new JAXBContextFactory.Builder()
+		.withMarshallerJAXBEncoding(StandardCharsets.UTF_8.toString())
+		.build();
+
+	private static final SOAPEncoder.Builder ENCODER_BUILDER = new SOAPEncoder.Builder()
+		.withCharsetEncoding(StandardCharsets.UTF_8)
+		.withFormattedOutput(false)
+		.withJAXBContextFactory(JAXB_FACTORY)
+		.withSOAPProtocol(SOAPConstants.SOAP_1_1_PROTOCOL)
+		.withWriteXmlDeclaration(true);
+
+	private final Map<ClientKey, OpeneClient> clients = new HashMap<>();
+	private final EncryptionUtility encryptionUtility;
+	private final ApplicationContext applicationContext;
+
+	OpeneClientFactory(final InstanceRepository instanceRepository, final EncryptionUtility encryptionUtility, final ApplicationContext applicationContext) {
+		this.encryptionUtility = encryptionUtility;
+		this.applicationContext = applicationContext;
+
+		instanceRepository.findByIntegrationType(SOAP).forEach(instanceEntity -> {
+			createSoapClient(instanceEntity);
```

Comment:
Should two SOAP clients be created?

Reply:
Nope. Should just create one client per configured instance.